datalad/tests/utils_pytest.py

```diff
@@ -1632,57 +1632,72 @@ def ignore_nose_capturing_stdout(func):
         OBSCURE_FILENAME_PARTS.append(UNICODE_FILENAME)
 # space before extension, simple extension and trailing space to finish it up
 OBSCURE_FILENAME_PARTS += [' ', '.datc', ' ']
+# Parts that should not be used in directory names
+DIRECTORY_EXCLUDED_PARTS = ['\n']
 
 
 @with_tempfile(mkdir=True)
-def get_most_obscure_supported_name(tdir, return_candidates=False):
+def get_most_obscure_supported_name(tdir, return_candidates=False, directory_name=False):
     """Return the most obscure filename that the filesystem would support under TEMPDIR
 
     Parameters
     ----------
     return_candidates: bool, optional
       if True, return a tuple of (good, candidates) where candidates are "partially"
       sorted from trickiest considered
+
+    directory_name: bool, optional
+      if True, return an obscure directory name. Directories might have
+      different obscurity profiles. Currently, directory names will not
+      contain newline characters.
+
     TODO: we might want to use it as a function where we would provide tdir
     """
     # we need separate good_base so we do not breed leading/trailing spaces
     initial = good = OBSCURE_PREFIX
     system = platform.system()
 
-    OBSCURE_FILENAMES = []
+    obscure_filenames = []
     def good_filename(filename):
-        OBSCURE_FILENAMES.append(candidate)
+        obscure_filenames.append(candidate)
         try:
-            # Windows seems to not tollerate trailing spaces and
+            # Windows seems to not tolerate trailing spaces and
             # ATM we do not distinguish obscure filename and dirname.
             # So here we will test for both - being able to create dir
             # with obscure name and obscure filename under
-            os.mkdir(opj(tdir, filename))
-            with open(opj(tdir, filename, filename), 'w') as f:
-                f.write("TEST LOAD")
+            if directory_name is True:
+                os.mkdir(opj(tdir, filename))
+            else:
+                with open(opj(tdir, filename), 'w') as f:
+                    f.write("TEST LOAD")
             return True
         except:
-            lgr.debug("Filename %r is not supported on %s under %s",
-                      filename, system, tdir)
+            lgr.debug(
+                f"{'Filename' if directory_name is False else 'Dirname'} %r is "
```

Comment:
May be
```suggestion
                f"{'Dir' if directory_name else 'File'}name %r is "
```
would be easier since shorter etc?